File: app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/teidata/DashboardProgramAdapter.java
package org.dhis2.usescases.teiDashboard.dashboardfragments.teidata;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import org.dhis2.R;
import org.dhis2.databinding.ItemDashboardProgramBinding;
import org.dhis2.usescases.teiDashboard.DashboardProgramModel;

/**
 * Created by ppajuelo on 27/02/2018.
 *
 */

public class DashboardProgramAdapter extends RecyclerView.Adapter<DashboardProgramViewHolder> {

    private final TEIDataContracts.Presenter presenter;
    private DashboardProgramModel dashboardProgramModel;

    public DashboardProgramAdapter(TEIDataContracts.Presenter presenter, DashboardProgramModel program) {
        this.dashboardProgramModel = program;
        this.presenter = presenter;
    }

    @NonNull
    @Override
    public DashboardProgramViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemDashboardProgramBinding binding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()), R.layout.item_dashboard_program, parent, false);
        return new DashboardProgramViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull DashboardProgramViewHolder holder, int position) {
        holder.bind(presenter, dashboardProgramModel, position);
    }

    @Override
    public int getItemCount() {
        return dashboardProgramModel.getEnrollmentPrograms().size();
    }
}

File: app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/teidata/DashboardProgramAdapter.kt
package org.dhis2.usescases.teiDashboard.dashboardfragments.teidata

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import org.dhis2.R
import org.dhis2.databinding.ItemDashboardProgramBinding
import org.dhis2.usescases.teiDashboard.DashboardProgramModel

class DashboardProgramAdapter(
    private val presenter: TEIDataPresenter,
    private val dashboardProgramModel: DashboardProgramModel
) : RecyclerView.Adapter<DashboardProgramViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DashboardProgramViewHolder {
        val binding = DataBindingUtil.inflate<ItemDashboardProgramBinding>(
            LayoutInflater.from(parent.context),
            R.layout.item_dashboard_program,
            parent,
            false
        )
        return DashboardProgramViewHolder(binding)
    }

    override fun onBindViewHolder(holder: DashboardProgramViewHolder, position: Int) {
        holder.bind(presenter, dashboardProgramModel, position)
    }

    override fun getItemCount(): Int {
        return dashboardProgramModel.programsWithActiveEnrollment.size
    }
}
